site/docs/guides/basic-drawing.md

```diff
@@ -322,8 +324,13 @@ function pixel (x: i32, y: i32): void {
     const shift = u8((x & 0b11) << 1);
     const mask = u8(0b11 << shift);
 
-    // Use the first DRAW_COLOR as the pixel color
-    const color = u8(load<u16>(w4.DRAW_COLORS) & 0b11);
+    // Use the first DRAW_COLOR as the pixel color.
+    const palette_color = u8(load<u16>(w4.DRAW_COLORS) & 0b1111);
```

Comment:
This still allows two large a range.. you'd really want to subtract by 1 first, then mask to `0b11`.

This would allow a color of 6 for example to slip thru...

Reply:
Subtracting 1 first could underflow. I guess that's probably ok in a lot of these, though is UB in zig.

Reply:
You're right though that we should do something to handle values > 4. I was thinking mask again with 0b11 after the subtraction.

Reply:
That would have strange behavior since your'e still allowing invalid data.  I'd rather strip the invalid data entirely.

To get it 100% right it seems you first need to check for 0 right off the bat and just return (do nothing)... after that it's as simple as -1, mask.